### PR TITLE
[ImageCard] add upload capabilities

### LINKED ISSUE #1754: [ImageCard] Add empty/upload state
Part of #1741 

As part of the dashboard card editor flow for ImageCard we need to add a new emtpy/upload state to the ImageCard

image

### PR DESCRIPTION
**Pull request by davidicus**
Closes #1754 
Part of #1817 

**Summary**

Add ability to upload images from local files or from url. Part of the Dashboard editor effort. 

**Change List (commits, features, bugs, etc)**

- created a new ImageUploader component
- create a new check in ImageCard for existence of `isNew` prop and serve up uploader
- created _image-uploader.scss 
-added new utility function to handle fetch and conversion of images from url in `cardUtilityFunctions.js`

**Acceptance Test (how to verify the PR)**

- Look at new story created for image uploader

### CODE REVIEW

**Review by scottdickerson — APPROVED** [review]
LGTM

File: src/components/ImageCard/ImageUploader.test.jsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    userEvent.click(
+      screen.getByText(/Drag and drop file here or click to select file/)
+    );
+    // await waitFor(() => expect(screen.getAllByText(/OK/)).toHaveLength(1));
+    fireEvent.change(container.querySelector('input'), {
+      target: { files: [file] },
+    });
+    // userEvent.click(screen.getByText(/OK/));
+    screen.debug();
```
Comment (tay1orjones) [inline src/components/ImageCard/ImageUploader.test.jsx:34]:
```suggestion
```

File: src/components/ImageCard/_image-uploader.scss [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  }
+
+  .#{$iot-prefix}--image-uploader-url-wrapper {
+    flex-direction: row;
+  }
+}
+
+// .#{$iot-prefix}--image-uploader.#{$iot-prefix}--image-uploader__large {
+// }
+
```
Comment (tay1orjones) [inline src/components/ImageCard/_image-uploader.scss:150]:
```suggestion
```

**Review by tay1orjones — APPROVED** [review]
Two minor items

**Pull request merged**